pandas/_typing.py

```diff
@@ -17,6 +17,16 @@
 
 import numpy as np
 
+try:
+    # PY38+
+    from typing import final
+except ImportError:
+    # define a dummy
+
+    def final(x):
```

Comment:
Should we use `functools.wraps`?

Reply:
also need to add types to the dummy to preserve the type declartions on the decorated functions, see https://mypy.readthedocs.io/en/stable/generics.html#declaring-decorators
```suggestion
    def final(x: F) -> F:
```

Reply:
> Should we use functools.wraps?

I don't think so, the returned function isnt a new function, just the old one unchanged